versions.go

```diff
@@ -30,13 +70,46 @@ type MongoDBVersion struct {
 	tag      string
 }
 
-// NewMongoDBVersion takes a string representing a MongoDB version and
-// returns a MongoDBVersion object. If the input string is not a valid
+// NewMongoDBVersion is a structure representing a version identifier for versions of
+// MongoDB, which implements the MongoDBVersion.
+type NewMongoDBVersion struct {
+	LegacyMongoDBVersion
+}
+
+func CreateMongoDBVersion(version string) (MongoDBVersion, error) {
+	endOfLegacyVersion, _ := semver.Parse(endOfLegacy)
+
+	// pre-processing to then determine which version to use
+	toParse := version
+	if strings.HasSuffix(version, "-") && !strings.Contains(version, "pre") {
+			toParse += "pre-"
+	}
+	if strings.Contains(toParse, "~") {
+		versionParts := strings.Split(version, "~")
+		toParse = versionParts[0]
+	}
+
+	parsed, err := semver.Parse(toParse)
+	if err != nil {
+		return nil, err
+	}
+	if parsed.LT(endOfLegacyVersion) {
+		return createLegacyMongoDBVersion(version)
+	}
+	return createNewMongoDBVersion(version)
+}
+
+func createNewMongoDBVersion(version string) (*NewMongoDBVersion, error) {
+	return nil, errors.New("not yet implemented")
+}
+
+
+// CreateMongoDBVersion takes a string representing a MongoDB version and
+// returns a LegacyMongoDBVersion object. If the input string is not a valid
 // version, or there were problems parsing the string, the error value
 // is non-nil. All parsing of a version happens during this phase.
-func NewMongoDBVersion(version string) (*MongoDBVersion, error) {
-	v := &MongoDBVersion{source: version, rcNumber: -1}
-
+func createLegacyMongoDBVersion(version string) (*LegacyMongoDBVersion, error) {
```

Comment:
Wrong func name in comment, also probably don't need documentation here since not exported.

Reply:
I would prefer to keep the documentation. It's not exported but oftentimes I find it useful to keep comments on unexported things if they ease understanding the code.

Reply:
I'm removing part of the documentation since it's actually just explaining what an error is